demos/multi_channel_object_detection_demo_yolov3/cpp/main.cpp

```diff
@@ -31,61 +31,44 @@
-    std::cout << "    -bs                          " << batch_size << std::endl;
-    std::cout << "    -nireq                       " << num_infer_requests << std::endl;
-    std::cout << "    -n_iqs                       " << input_queue_size << std::endl;
-    std::cout << "    -fps_sp                      " << fps_sampling_period << std::endl;
-    std::cout << "    -n_sp                        " << num_sampling_periods << std::endl;
-    std::cout << "    -t                           " << thresh_output_message << std::endl;
-    std::cout << "    -no_show                     " << no_show_message << std::endl;
-    std::cout << "    -show_stats                  " << show_statistics << std::endl;
-    std::cout << "    -real_input_fps              " << real_input_fps << std::endl;
-    std::cout << "    -u                           " << utilization_monitors_message << std::endl;
-}
-
-bool ParseAndCheckCommandLine(int argc, char *argv[]) {
-    // ---------------------------Parsing and validation of input args--------------------------------------
-    gflags::ParseCommandLineNonHelpFlags(&argc, &argv, true);
-    if (FLAGS_h) {
-        showUsage();
+constexpr char threshold_message[] = "Probability threshold for detections";
+DEFINE_double(t, 0.5, threshold_message);
+
+void parse(int argc, char *argv[]) {
+    gflags::ParseCommandLineFlags(&argc, &argv, false);
+    slog::info << ov::get_openvino_version() << slog::endl;
+    if (FLAGS_h || argc == 1) {
+        std::cout << "\n    [-h]              " << help_message
+                  << "\n     -i               " << input_message
+                  << "\n    [-loop]           " << loop_message
+                  << "\n    [-duplicate_num]  " << duplication_channel_number_message
+                  << "\n     -m <path>        " << model_path_message
+                  << "\n    [-d <device>]     " << target_device_message
+                  << "\n    [-bs]             " << batch_size
+                  << "\n    [-n_iqs]          " << input_queue_size
+                  << "\n    [-fps_sp]         " << fps_sampling_period
+                  << "\n    [-n_sp]           " << num_sampling_periods
+                  << "\n    [-t]              " << threshold_message
+                  << "\n    [-no_show]        " << no_show_message
+                  << "\n    [-show_stats]     " << show_statistics
+                  << "\n    [-real_input_fps] " << real_input_fps
+                  << "\n    [-u]              " << utilization_monitors_message;
         showAvailableDevices();
-        return false;
+        std::exit(0);
+    } if (FLAGS_m.empty()) {
+        throw std::runtime_error("Parameter -m is not set");
+    } if (FLAGS_i.empty()) {
+        throw std::runtime_error("Parameter -i is not set");
+    } if (FLAGS_duplicate_num == 0) {
+        throw std::runtime_error("Parameter -duplicate_num must be positive");
+    } if (FLAGS_bs != 1) {
```

Comment:
what's is the point to allow set batch, but limit it to 1?

Reply:
Other 2 multi_channel demos support it. Yolo postprocessing was written to support only batch 1. I just made it explicit.

Reply:
Maybe better would be to remove this flag and add message about setting batch to one or smth like this, what do you think?

Reply:
I can't remove it. It is defined in multi_channel_common

Reply:
Remove from help message then?

Reply:
It's just kind of strange from user's perspective to have ability to set batch according to help message, but when you try it will say that just one value support

Reply:
Good point. Removed